runtime/serde/serde-json/TESTING.md

```diff
@@ -0,0 +1,178 @@
+How to run JSONTestSuite against serde-json deserialize
+========================================================
+
+When making changes to the lexer it is a good idea to run the
+changes against the [JSONTestSuite](https://github.com/nst/JSONTestSuite) and manually examine the test results.
+
+### How to setup the JSONTestSuite
+
+1. Clone the [JSONTestSuite](https://github.com/nst/JSONTestSuite) repository.
+2. In `JSONTestSuite/parsers`, create a new Gradle JVM application project named `test_smithy_kotlin`.
+3. Add the following `build.gradle.kts` file
+
+```kotlin
+plugins {
+    kotlin("jvm") version "1.5.30"
+    application
+    id("com.github.johnrengelman.shadow") version "7.0.0"
+}
+
+application {
+    mainClass.set("aws.smithy.kotlin.jsontest.MainKt")
+}
+
+allprojects {
+    repositories {
+        mavenLocal()
+        mavenCentral()
+    }
+}
+
+
+// NOTE: set to whatever locally published version you are working on
+val smithyKotlinVersion: String = "0.4.1-kmp-json"
+dependencies {
+   implementation(kotlin("stdlib"))
+   implementation("org.jetbrains.kotlinx:kotlinx-coroutines-core:1.5.0")
+   implementation("aws.smithy.kotlin:serde-json:$smithyKotlinVersion")
+   implementation("aws.smithy.kotlin:utils:$smithyKotlinVersion")
+}
+
+tasks.jar {
+    manifest {
+        attributes["Main-Class"] = "aws.smithy.kotlin.jsontest.MainKt"
+    }
+}
+```
+
+4. Add the following code to `src/main/kotlin/Main.kt` with:
+
+```kotlin
+package aws.smithy.kotlin.jsontest
```

Comment:
It isn't really important since this is creating an application jar. Also technically it would be in line with the recommended [directory structure](https://kotlinlang.org/docs/coding-conventions.html#directory-structure) if `aws.smithy.kotlin.jsontest` is the "root".